libqaul/common/src/message.rs

```diff
@@ -0,0 +1,19 @@
+//! Types for representing a message to be sent between users.
+use crate::UserID;
+use crate::payload::Payload;
+
+/// A `Message` to be sent from the `sender` to the `recipient`, associated
+/// with the given `service`.
+/// Differs from a RATMAN message; this is a user-to-user, fully encapsulated
+/// message which does not care (or know) how it is delivered, merely that it
+/// was sent by one user and is meant for another.
+pub struct Message {
+    sender: UserID,
+    recipient: UserID,
+    // TODO DESIGN: Include a fingerprint here?
+    // TODO DESIGN: It's probably a good idea to add a signature in addition to
```

Comment:
I would say "yes" to both of those.

Although firstly I guess we should answer the question: is the fingerprint and "UserID" different? From how I see it we have three IDs at play here

- qaul.net ID
- fingerprint
- RATMAN routing ID

We could just make all three the same. The potential upside is that it requires less space. Not sure if there's any downsides other than _maybe_ more likely collisions?

Once we figured this out, I would suggest we include the signature of some always-changing payload like a timestamp or message counter

Reply:
I definitely think we should collapse as much as possible, but the qaul.net ID and the RATMAN ID are conceptually different, right?

Reply:
Funnily enough, not as much as you'd think.  RATMAN doesn't actually have the concept of a "computer node" in its routing. It only cares about mapping ids to specific network addresses. And those ids could very well just be the user ids i.e. user fingerprints.  This also means that if someone wanted to get a new id on the network, they could easily do this and pretty much nobody on the network would be able to figure out that they're the same person

(not sure what the email reply did here first 😅)

Reply:
OK, let's make them all the same.